packages/jest-diff/src/cli/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import yargs from 'yargs';
+import {readFileSync} from 'fs';
+
+// eslint-disable-next-line import/default
+import jestDiff from '../';
+import {NO_DIFF_MESSAGE} from '../constants';
```

Comment:
This is IMO an example of where a separate CLI would force a better API.

It's imported to allow the CLI to remove a message that shouldn't really be in a diffing algorithm. Instead it should probably be an option (or structured return values, or something else entirely), but since we have access to private stuff, it's more natural to just import it

Reply:
Yeah it really bugged me writing this that the return values of jest-diff are so weird :+1:

Reply:
Yeah, those constants :(

Reply:
Can we fix this without a breaking change?

Reply:
Yeah, adding an option which changes the returned thing (from a string to something structured) should be fine in a minor (as long as it's opt-in). Might be a bit painful WRT flow types but probably fine

Reply:
e.g. returning something like `{ pretty(): string, hasDiff: boolean }` where jest calls `pretty()`. @pedrottimark probably has way better API ideas than that, but it's an idea

Reply:
The existing `null` return values related to asymmetric matchers are also quite a pain point. Hope Mark saves us with an amazing idea :smile:

Reply:
Thank you, mee too :/

Reply:
Regarding the `null` return values for asymmetric matchers: Would it be fine to just treat them like "no difference"? The reason they were originally introduced was https://github.com/facebook/jest/issues/8146 - in that case, you would get the "No visual difference" message as well. Sooner or later we'll have to rethink `jest-diff` (and `expect` and `pretty-format`) in light of https://github.com/facebook/jest/issues/6184 , but for now it would be good to have just "differs, here's the string" and "does not differ (visually)" as returns values so that we can come up with a clean and simple API

Reply:
(Actually, what's  now `SIMILAR_MESSAGE` will need to have a special case in the return value too :/ )

Reply:
Perhaps we should return something like `{differs: true, message: string, toJsonIgnored: boolean} | {differs: false}`